core/ledger/kvledger/kv_ledger.go

```diff
@@ -984,10 +989,17 @@ func (a *ccEventListenerAdaptor) ChaincodeDeployDone(succeeded bool) {
 	a.legacyEventListener.ChaincodeDeployDone(succeeded)
 }
 
-func filterPvtDataOfInvalidTx(hashVerifiedPvtData map[uint64][]*ledger.TxPvtData, blockStore *blkstorage.BlockStore) (map[uint64][]*ledger.TxPvtData, error) {
+func filterPvtDataOfInvalidTx(
+	hashVerifiedPvtData map[uint64][]*ledger.TxPvtData,
+	blockStore *blkstorage.BlockStore,
+	lastBlockInBootstrapSnapshot uint64,
+) (map[uint64][]*ledger.TxPvtData, error) {
 	committedPvtData := make(map[uint64][]*ledger.TxPvtData)
 	for blkNum, txsPvtData := range hashVerifiedPvtData {
-
+		if blkNum <= lastBlockInBootstrapSnapshot {
+			committedPvtData[blkNum] = txsPvtData
+			continue
+		}
```

Comment:
ledger level integration test can be added in a separate PR along with other remaining tests.